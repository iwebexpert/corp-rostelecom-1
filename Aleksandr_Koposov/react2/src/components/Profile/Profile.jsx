import React, { useEffect, useState } from 'react'

import {
  Card,
  CardContent,
  CardActions,
  Badge,
  Icon,
  Avatar,
  TextField,
  Button,
  Dialog,
  DialogTitle,
  CircularProgress
} from '@material-ui/core'

import './Profile.scss'

export const Profile = (props) => {
  const { user, isLoading } = props
  const { id, name, age, about, avatar } = user || {}
  const [profileState, setProfile] = useState(
    {
      init: false,
      dialog: false,
      id,
      name,
      age,
      about,
      avatar
    }
  )

  useEffect(() => {
    setProfile({
      ...profileState,
      init: true,
      ...props.user
    })
  }, [])

  const onInputChange = (event) => {
    setProfile({
      ...profileState,
      [event.target.name]: event.target.value
    })
  }

  const saveChanges = () => {
    const { onSave } = props
    const { id, name, age, about, avatar } = profileState
    onSave({ id, name, age, about, avatar })
  }

  const toggleDialog = () => {
    setProfile({
      ...profileState,
      dialog: !profileState.dialog
    })
  }

  const avatarSize = {
    width: 128,
    height: 128
  }

  return (
    <Card className="profile" elevation={3}>
      <CardContent>
        <Badge
          overlap="circle"
          anchorOrigin={{
            vertical: 'bottom',
            horizontal: 'right',
          }}
          badgeContent={<Icon>edit</Icon>}
          onClick={toggleDialog}
        >
          {!isLoading ? (
            <Avatar
              style={avatarSize}
              alt="Аватар пользователя"
              src={profileState.avatar}
            />
          ) : (
              <Avatar style={avatarSize}>
                <CircularProgress />
              </Avatar>
            )}
        </Badge>
        <Dialog
          onClose={toggleDialog}
          open={profileState.dialog}
          className="avatar-dialog"
        >
          <DialogTitle>Выбор аватара</DialogTitle>
          <Card>
            <CardContent>
              <TextField
                name="avatar"
                label="Ссылка на аватар (можете изменять число в конце для теста)"
                variant="outlined"
                value={profileState.avatar}
                onChange={onInputChange}
                style={{ width: '550px' }}
              />
            </CardContent>
          </Card>
        </Dialog>
        {!isLoading ?
          (
            <div>
              <TextField
                name="name"
                label="Имя"
                variant="outlined"
                value={profileState.name}
                onChange={onInputChange}
              />
              <TextField
                name="age"
                label="Возраст"
                variant="outlined"
                type="number"
                value={profileState.age}
                onChange={onInputChange}
              />
              <TextField
                name="about"
                label="О себе"
                multiline
                variant="outlined"
                value={profileState.about}
                onChange={onInputChange}
              />
            </div>
          ) : (
            <div style={{
              width: '100%',
              textAlign: 'center'
            }}>
              <CircularProgress />
            </div>
          )}
      </CardContent>
      <CardActions>
        <Button
          color="primary"
          onClick={saveChanges}
        >Сохранить</Button>
      </CardActions>
    </Card>
  )
}
