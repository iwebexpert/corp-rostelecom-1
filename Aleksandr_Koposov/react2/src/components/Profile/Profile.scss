.profile {
  & .MuiBadge-badge {
    height: 36px;
    width: 36px;
    border-radius: 50%;
    background-color: grey;
    color: white;
    cursor: pointer;
  }

  & .MuiTextField-root {
    width: 100%;
    margin: 15px 0;
  }
}

.avatar-dialog {
  & .MuiTextField-root {
    max-width: 550px;
    width: 100%;
  }
}
