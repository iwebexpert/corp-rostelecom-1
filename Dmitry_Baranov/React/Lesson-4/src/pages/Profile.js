import React from 'react';

export function ProfilePage(){
    return (
        <div>
            <h3>Profile page</h3>
            <p><i>Имя: </i>Имя пользователя</p>
            <p><i>Ник: </i>Псевдоним пользователя</p>
            <p><i>Возраст: </i>Возраст пользователя</p>
        </div>
    );
}