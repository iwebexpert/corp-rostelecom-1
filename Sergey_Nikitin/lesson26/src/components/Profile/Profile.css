.chatList {
    width: '30%';
    max-width: 360;
    background-color: paper;
}

.MuiListItem-root:hover {
    background-color: rgba(0, 0, 0, 0.02);
    cursor: grab;
}

.root {
    max-width: 345px;
    margin-left: 100px;
}