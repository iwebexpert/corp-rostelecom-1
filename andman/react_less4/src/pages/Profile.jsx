import React from 'react';

export function ProfilePage() {
    return (
        <div>
            <h2>Profile page</h2>
            E-mail:  test@mail.ru <hr />
            Логин: Test1 <hr />
        </div>


    );
}