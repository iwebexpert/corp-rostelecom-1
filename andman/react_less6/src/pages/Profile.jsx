import React, { Component } from 'react';

export class ProfilePage extends Component {

    render() {
        const { profile } = this.props;
        //console.log(profile);
        return (
            <div>
                <h2>Profile page</h2>
        E - mail: {profile.email} <hr />
        Логин: {profile.name} < hr />
                <img src={profile.avatar} width="200px" height="200px" />
            </div >
        );
    };
}