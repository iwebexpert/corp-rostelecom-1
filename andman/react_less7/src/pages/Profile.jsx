import React, { Component } from 'react';

export class ProfilePage extends Component {

    render() {

        const { profile, isLoading, isError } = this.props;
        if (isError) {
            return (
                <div>Не удалось загрузить профиль...</div>
            );
        }
        if (isLoading) {
            return (
                <div>Profile loading...</div>
            );
        }
        return (
            <div>
                <h2>Profile page</h2>
        E - mail: {profile.email} <hr />
        Логин: {profile.name} < hr />
                <img src={profile.avatar} width="200px" height="200px" />
            </div >
        );

    };
}