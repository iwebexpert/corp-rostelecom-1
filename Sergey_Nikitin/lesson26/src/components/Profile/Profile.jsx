import React, { Component } from 'react';
import Card from '@material-ui/core/Card';
import CardActionArea from '@material-ui/core/CardActionArea';
import CardActions from '@material-ui/core/CardActions';
import CardContent from '@material-ui/core/CardContent';
import CardMedia from '@material-ui/core/CardMedia';
import Button from '@material-ui/core/Button';
import Typography from '@material-ui/core/Typography';

import CircularProgress from '@material-ui/core/CircularProgress';

import { Grid, Container, List, ListItem, ListItemText } from '@material-ui/core';

import './Profile.css';

export class Profile extends Component {

    render() {
        const profile = this.props.profile;
        const { isLoading, isError, handleReloadProfile } = this.props;
        if (isError) {
            return (<Grid container direction="row" justify="center">
                <Grid item>
                    <div>Не удалось загрузить... <button onClick={handleReloadProfile}>Загрузить повторно</button></div></Grid>
            </Grid>);
        }

        if (isLoading) {

            return (
                <Grid container direction="row" justify="center">
                    <Grid item>
                        <CircularProgress />
                    </Grid>
                </Grid>);

        }
        return (
            <Grid container direction="row" justify="center">
                <Grid item>
                    <Card className="root">
                        <CardActionArea>
                            <CardMedia
                                component="img"
                                alt="Contemplative Reptile"
                                height="140"
                                image={profile.image}
                                title="Contemplative Reptile"
                            />
                            <CardContent>
                                <Typography gutterBottom variant="h5" component="h2">
                                    {profile.email}
                                </Typography>
                                <Typography variant="body2" color="textSecondary" component="p">
                                    {profile.about}
                                </Typography>
                            </CardContent>
                        </CardActionArea>
                        <CardActions>
                            <Button size="small" color="primary">
                                Share
        </Button>
                            <Button size="small" color="primary">
                                Learn More
        </Button>
                        </CardActions>
                    </Card>
                </Grid>
            </Grid>
        );
    }
}
