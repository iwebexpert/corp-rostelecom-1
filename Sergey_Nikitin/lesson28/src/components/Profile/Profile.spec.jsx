import { shallow } from 'enzyme';
import React from 'react';
import { Profile } from './Profile';

//it
test('Simple test', () => {
    const a = 3;
    expect(a + 2).toBe(5);
});

test('should be rendered Profile', () => {
    const prof = {
        "title": 'Chat1',
        "email": "Test@test.ru",
        "about": "TEST",
        "image": "https://img2.wtftime.ru/store/2020/09/08/6ZW02821.jpg"
    };
    const component = shallow(<Profile profile={prof} handleReloadProfile={null}
        isError={false}
        isLoading={false} />);
    /*
     console.log(component);
      const wrapper1 = component.find('Typography');
      console.log(wrapper1.length);*/
    /* expect(wrapper1.length).toBe(3);
     expect(wrapper1.last().text()).toBe('TEST');
     expect(wrapper1.first().text()).toBe('Test@test.ru');*/
    expect(component).toMatchSnapshot();
});

