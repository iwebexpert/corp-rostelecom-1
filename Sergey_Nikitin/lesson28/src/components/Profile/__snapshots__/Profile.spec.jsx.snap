// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should be rendered Profile 1`] = `
<WithStyles(ForwardRef(Grid))
  container={true}
  direction="row"
  justify="center"
>
  <WithStyles(ForwardRef(Grid))
    item={true}
  >
    <WithStyles(ForwardRef(Card))
      className="root"
    >
      <WithStyles(ForwardRef(CardActionArea))>
        <WithStyles(ForwardRef(CardMedia))
          alt="Contemplative Reptile"
          component="img"
          height="140"
          image="https://img2.wtftime.ru/store/2020/09/08/6ZW02821.jpg"
          title="Contemplative Reptile"
        />
        <WithStyles(ForwardRef(CardContent))>
          <WithStyles(ForwardRef(Typography))
            component="h2"
            gutterBottom={true}
            variant="h5"
          >
            Test@test.ru
          </WithStyles(ForwardRef(Typography))>
          <WithStyles(ForwardRef(Typography))
            color="textSecondary"
            component="p"
            variant="body2"
          >
            TEST
          </WithStyles(ForwardRef(Typography))>
        </WithStyles(ForwardRef(CardContent))>
      </WithStyles(ForwardRef(CardActionArea))>
      <WithStyles(ForwardRef(CardActions))>
        <WithStyles(ForwardRef(Button))
          color="primary"
          size="small"
        >
          Share
        </WithStyles(ForwardRef(Button))>
        <WithStyles(ForwardRef(Button))
          color="primary"
          size="small"
        >
          Learn More
        </WithStyles(ForwardRef(Button))>
      </WithStyles(ForwardRef(CardActions))>
    </WithStyles(ForwardRef(Card))>
  </WithStyles(ForwardRef(Grid))>
</WithStyles(ForwardRef(Grid))>
`;
